USSF-ORBIT/ussf-portal-client
# dev: Fetch example collection from Keystone

# SC-293


## Proposed changes

* Queries Keystone CMS for example collection data when creating a new user

**To Do**
- [x] Fix error when adding example collection to new user, add back Collection.addOne

## Reviewer notes

- Test various new user creation scenarios and make sure the example collection is added as expected.
- Run both the portal & the CMS locally. Make a change to the Example Collection in the CMS (change the title, add a new link, etc.) before signing up as a new user, and verify that when you do, the Example Collection is added to that user with the expected changes.

## Setup

To test user creation:

* `yarn services:up` - Start required services
* `yarn dev` - Start portal
* In ussf-portal-cms, run `yarn dev` - Start cms
* Visit `localhost:8888` > Click `dev` > Delete test users
* Log in as test users represented here - https://github.com/USSF-ORBIT/ussf-portal-client/blob/main/docs/development.md#logging-in

This should trigger a new user creation in MongoDB and add the example collection.

You can also edit the Example Collection in the CMS, add another new user, and observe that the correct updated links are added. Note: Because the example collection is added to a user's data and is not strictly a reference to the CMS, editing it will not impact users whose accounts were created with the previous version.

## Code review steps

### As the original developer, I have

  - ... 
  - [ ] Including cypress-axe checks when UI changes
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
- For any new migrations/schema changes:

### As code reviewer(s), I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a designer reviewer, I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11 (redirects to /update-browser)
  - [ ] On commercial internet in Edge
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR
